aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -117,4 +117,20 @@ Tensor to_mkldnn_backward(const Tensor& grad, const Tensor& input_) {
   return grad.to_dense(input_.scalar_type());
 }
 
+Tensor view_dtype(const Tensor& self, ScalarType dtype) {
+  if (self.scalar_type() == dtype) {
+    return self;
+  }
+  auto type_meta = c10::scalarTypeToTypeMeta(dtype);
+  TORCH_CHECK(self.element_size() == type_meta.itemsize(),
+    "Viewing a tensor as a new dtype with a different number of bytes per element is not supported.");
+  Storage storage = self.storage();
+  auto new_tensor = detail::make_tensor<TensorImpl>(
+      std::move(storage), self.key_set(), type_meta);
```

Comment:
@zasdfgbnm do you know what would happen if original tensor required grad, and you are viewing it as integer? Would key_set still have autograd key? The operation is non-differentiable, so it's not particularly important, unless it crashes or produces confusing error message. Can you add a test for what would happen?

Reply:
It will just return a tensor with requires_grad=False